smartsim/_core/generation/generator.py

```diff
@@ -24,53 +24,116 @@
 # OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+import base64
+import os
 import pathlib
+import pickle
 import shutil
 import typing as t
 from datetime import datetime
 from distutils import dir_util  # pylint: disable=deprecated-module
```

Comment:
Side thing: `disutils` was [deprecated in 3.10 and has been removed in 3.12](https://peps.python.org/pep-0632/). Can you do me a solid and remove this import!

Reply:
will do!